cloud/server.go

```diff
@@ -293,7 +294,12 @@ func (s *Server) Matches(os string, script []byte, configFiles string, flavor *F
 // there was enough space. Returns true if there was enough space and the
 // allocation occurred.
 func (s *Server) Allocate(ctx context.Context, cores float64, ramMB, diskGB int) bool {
-	ctx = clog.ContextWithServerID(ctx, s.ID)
+	if s.Name == localhostName {
```

Comment:
This shouldn't just be for Allocate and Release, and the code block shouldn't be duplicated. All current calls to clog.ContextWithServerID should instead be replaced with a call to a new method s.ContextWithServerID, which calls clog.ContextWithServerID(ctx, id), where id is localhostName if s.Name is localhostName.